cloud/aws/services/ec2/vpc_test.go

```diff
@@ -14,37 +14,79 @@
 package ec2_test
 
 import (
+	"errors"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/providerconfig/v1alpha1"
 	ec2svc "sigs.k8s.io/cluster-api-provider-aws/cloud/aws/services/ec2"
 )
 
 type vpcs struct{}
 
-func (v *vpcs) DescribeVpcs(i *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error) {
-	return &ec2.DescribeVpcsOutput{
-		Vpcs: []*ec2.Vpc{
-			&ec2.Vpc{
-				VpcId: aws.String("world"),
-			},
+func (v *vpcs) CreateVpc(i *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error) {
+	return &ec2.CreateVpcOutput{
+		Vpc: &ec2.Vpc{
+			VpcId:     aws.String("newVpc"),
+			CidrBlock: aws.String("10.0.0.0/16"),
 		},
 	}, nil
 }
 
+func (v *vpcs) DeleteVpc(i *ec2.DeleteVpcInput) (*ec2.DeleteVpcOutput, error) {
+	return &ec2.DeleteVpcOutput{}, nil
+}
+
+func (v *vpcs) DescribeVpcs(i *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error) {
+	if len(i.VpcIds) == 0 {
+		return nil, errors.New("invalid request")
+	}
+
+	if *i.VpcIds[0] == "oldVpc" {
+		return &ec2.DescribeVpcsOutput{
+			Vpcs: []*ec2.Vpc{
+				&ec2.Vpc{
+					VpcId:     aws.String("oldVpc"),
+					CidrBlock: aws.String("10.0.0.0/16"),
+				},
+			},
+		}, nil
+	}
+
+	return &ec2.DescribeVpcsOutput{}, nil
+}
+
 func TestReconcileVPC(t *testing.T) {
 	s := ec2svc.Service{
 		Instances: nil,
 		VPCs:      &vpcs{},
 	}
 
-	id := "world"
-	vpc, err := s.ReconcileVPC(id)
-	if err != nil {
-		t.Fatalf("got an unexpected error: %v", err)
-	}
-	if vpc.ID != "world" {
-		t.Fatalf("Expected an id of %v but found %v", "world", vpc.ID)
-	}
+	t.Run("WithExistingVPC", func(t *testing.T) {
```

Comment:
please wrap these into a table test like this:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/67/files#diff-76dfa0346698cdb9b60bf6a38a07141bR71

Generally each error case will require its own TestFunction or there will be a ton of extra boolean flags and complex logic in the t.Run() function. Or the error cases can be untested if it's not worth it. That's a trade off I often make.